android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -79,24 +79,26 @@ private boolean isReactApplication(Context context) {
         return false;
     }
 
-    private void loadBundleLegacy() {
-        Activity currentActivity = getCurrentActivity();
-        Intent intent = currentActivity.getIntent();
-        currentActivity.finish();
-        currentActivity.startActivity(intent);
-
+    private void loadBundleLegacy(final Activity currentActivity) {
         mCodePush.invalidateCurrentInstance();
+
+        currentActivity.runOnUiThread(new Runnable() {
```

Comment:
Yeah I think doing this check might makes sense, but it probably should be within `loadBundle`, as do the error check up front before propagating it further.